### PR TITLE
Feature/edit target

### PR DESCRIPTION
**Pull request by gabrielbursztein2**
#### Notes
  * Edit endpoint is not developed in the backend
  * When deleting a target the API always returns `Record not found` even though the record is in fact deleted, so I can't remove that target from the reducer when I get the response.

## Platforms tested:
- [x] iOS
- [ ] Android

delete-target

### CODE REVIEW

File: src/components/targets/TargetForm/index.js [hunk 1]
```diff
@@ -50,10 +58,23 @@ TargetForm.navigatorStyle = {
...
+    radius: state.getIn(['targets', 'targetRadius']),
+    title: state.getIn(['targets', 'targetTitle']),
+    topic: state.getIn(['targets', 'targetTopic'])
+  }
+});
+
+TargetForm = reduxForm({
   form: 'target',
-  validate: constraints.validations(constraints.targets)
+  validate: constraints.validations(constraints.targets),
```
Comment (agustito37) [inline src/components/targets/TargetForm/index.js]: Alo! You can destructure this when you import contraints.

File: src/containers/MainScreen/index.js [hunk 2]
```diff
@@ -85,26 +157,28 @@ class MainScreen extends Component {
 
     return (
       <View style={styles.container}>
-        <TopBar title="Target Points" />
         {loading
           ? <ActivityIndicator size="large" />
           : <Fragment>
               <MapView
```
Comment (agustito37) [inline src/containers/MainScreen/index.js]: You probably want to create a component with the MapView and all the logic you created for it, so you can reuse it later in another screen.

File: src/actions/targetActions.js [hunk 3]
```diff
@@ -50,3 +51,45 @@ export const loadTargets = () =>
         throw err;
       });
   };
+
+export const selectedTarget = target => ({
+  type: types.SELECTED_TARGET,
+  target
+});
+
+export const selectTarget = ({ target }) =>
```
Comment (kevquincke) [inline src/actions/targetActions.js]: The use of redux-thunk here doesn't seem necessary, it's more useful if you have an api call or something like that, but here you can move the dispatch action to the `mapDispatch` and call selectedTarget instead

File: src/actions/targetActions.js [hunk 4]
```diff
@@ -50,3 +51,45 @@ export const loadTargets = () =>
...
+  dispatch =>
+    dispatch(selectedTarget(target));
+
+export const deselectedTarget = () => ({
+  type: types.DESELECT_TARGET
+});
+
+export const deselectTarget = () =>
```
Comment (kevquincke) [inline src/actions/targetActions.js]: Same here

File: src/actions/targetActions.js [hunk 5]
```diff
@@ -50,3 +51,45 @@ export const loadTargets = () =>
...
+
+export const deleteTarget = () => ({
+  type: types.DELETE_TARGET
+});
+
+export const deleteTargetSuccess = () => ({
+  type: types.DELETE_TARGET_SUCCESS
+});
+
+export const deleteTargetFail = () => ({
```
Comment (kevquincke) [inline src/actions/targetActions.js]: I would name it failure
```suggestion
export const deleteTargetFailure = () => ({
```
Same for the action type

File: src/containers/MainScreen/index.js [hunk 6]
```diff
@@ -38,23 +48,76 @@ class MainScreen extends Component {
...
     });
   }
 
-  hideTargetForm = () => {
-    this.setState({
-      showTargetForm: false
+  onMarkerPress = (event: any) => {
+    const { selectTarget } = this.props;
+    const { latitude, longitude } = event.nativeEvent.coordinate;
+    const target = this.findTarget(latitude, longitude);
```
Comment (kevquincke) [inline src/containers/MainScreen/index.js]: You can destructure `target` here

File: src/containers/MainScreen/index.js [hunk 7]
```diff
@@ -38,23 +48,76 @@ class MainScreen extends Component {
...
+
   showTargetForm = () => {
     this.setState({
       showTargetForm: true
     });
   }
 
+  zoomCamera = (lat, lng, zoomIn = true, zoomAltitude = 0) => {
+    const { altitude } = this.state;
+    this.setState({
```
Comment (kevquincke) [inline src/containers/MainScreen/index.js]: Maybe you can put this in one line

File: src/containers/MainScreen/index.js [hunk 8]
```diff
@@ -65,17 +128,26 @@ class MainScreen extends Component {
         longitude: target.lng
       };
       return (
-        <TargetMarker key={target.id} coordinate={coordinate} radius={target.radius} topicId={target.topicId} zoom={zoom} />
+        <TargetMarker
+          key={target.id}
+          coordinate={coordinate}
+          radius={target.radius}
+          topicId={target.topicId}
```
Comment (kevquincke) [inline src/containers/MainScreen/index.js]: Can you destructure target here?

File: src/reducers/targetsReducer.js [hunk 9]
```diff
@@ -15,7 +19,10 @@ export const targetsReducer = (state = initialState, action) => {
     }
 
     case types.NEW_TARGET_SUCCESS: {
-      return state.set('loading', false);
+      const { target } = action;
+      const newTargets = state.get('targets').concat(Immutable.fromJS({ target }));
+      const newState = state.set('targets', Immutable.List(newTargets));
```
Comment (kevquincke) [inline src/reducers/targetsReducer.js]: I think `newTargets` is already an Immutable List

File: src/reducers/targetsReducer.js [hunk 10]
```diff
@@ -36,6 +43,36 @@ export const targetsReducer = (state = initialState, action) => {
...
+    }
+
+    case types.DELETE_TARGET: {
+      return state.set('loading', true);
+    }
+
+    case types.DELETE_TARGET_SUCCESS: {
+      const { id } = action;
+      const newTargets = state.get('targets').filter(target => target.id !== id);
+      const newState = state.set('targets', Immutable.List(newTargets));
```
Comment (kevquincke) [inline src/reducers/targetsReducer.js]: Same here

File: src/actions/targetActions.js (hunk 3 above)
Comment (gabrielbursztein2) [reply src/actions/targetActions.js]: Done

**Pull request merged**